README.md

```diff
@@ -1,2 +1,138 @@
-This plugin allows to define build environment as a Dockerfile, stored in project SCM.
-see also https://wiki.jenkins.io/display/JENKINS/Docker+Custom+Build+Environment+Plugin
+# Docker Custom Build Environment Plugin for Jenkins
+
+[![Jenkins Plugin](https://img.shields.io/jenkins/plugin/v/docker-custom-build-environment.svg)](https://plugins.jenkins.io/docker-custom-build-environment/)
+[![Jenkins Plugin Installs](https://img.shields.io/jenkins/plugin/i/docker-custom-build-environment.svg?color=blue)](https://plugins.jenkins.io/docker-custom-build-environment/)
+
+This plugin allows the definition of a build environment for a job using a Docker container.
+
+<img src="docs/images/docker-jenkins.png" width="300">
+
+## Introduction
+
+A major requirements for a Jenkins-based Continuous Integration/Delivery
+setup are:
+
+- Ensure the build/test environment is exactly reproducible from a pre-defined setup.
+- Ensure the build/test environment is well isolated from other builds executed on the same infrastructure.
+
+Docker is a great way to bootstrap reproducible and isolated environments. Compared to a virtual machine, it's faster to launch and lighter to use. Another benefit is the Docker image can be defined both as a binary image pulled from repository, or as a plain text Dockerfile you can store in SCM along side the project source code, so that the source and environment are always in sync and recorded.
+
+CloudBees Docker Custom Build Environment Plugin has been designed to make Docker Images / [Dockerfile](https://docs.docker.com/reference/builder/) a first class citizen in a continuous delivery setup, which allows the simplest way for a development team to manage the exact bits of the build environment whilst the infrastructure team only has to focus on available resources hosting arbitrary docker containers.
+
+## User Guide
+
+CloudBees Docker Custom Build Environment Plugin can be used from any job type, it appears in the "build environment" section and let you configure your build to run inside a Docker container.
+
+You can :
+
+- Select the Docker image to run the build as a **Docker image** to be pulled. This is comparable to the [docker-plugin](https://wiki.jenkins.io/display/JENKINS/Docker+Plugin) approach to offer docker agents within Jenkins, but without any prerequisites on the Docker image nor need for Administrator privileges to configure the adequate agent template.
```

Comment:
Would prefer that we refer to the plugins site rather than the wiki for references to other plugins.
```suggestion
- Select the Docker image to run the build as a **Docker image** to be pulled. This is comparable to the [docker-plugin](https://plugins.jenkins.io/docker-plugin/) approach to offer docker agents within Jenkins, but without any prerequisites on the Docker image nor need for Administrator privileges to configure the adequate agent template.
```

Reply:
ah, i will fix.